### PR TITLE
EdgeHub: Fix Mqtt message address parsing

### PR DESCRIPTION
**Pull request by varunpuranik**
Issue: Because of the handling of an optional trailing '/', a message address can sometimes match 2 templates. For example, a message address - a/foo/b/ matches both the following templates 
"a/{p1}/b/" and "a/{p1}/b/{p2}, with the value of p2 in the second case being "".

Fix: 
If there are multiple matches, then select the first match. Because of this, the order of the input templates matter (I confirmed they are in the right order).

Eventually we should move away form this and to a model based on Regex matching, where we won't have this problem. But that is a much bigger change, so will put it on our backlog.

### CODE REVIEW

File: edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Mqtt/MessageAddressConverter.cs [hunk 1]
```diff
@@ -86,32 +86,38 @@ public bool TryBuildProtocolAddressFromEdgeHubMessage(string endPointUri, IMessa
...
-            else if (matches.Count == 0)
+            if (matches.Count == 0)
             {
                 this.logger.LogWarning($"Message address {message.Address} does not match any topic.");
             }
             else
             {
+                if (matches.Count > 1)
+                {
+                    this.logger.LogDebug($"Message address {message.Address} matches more than one topic. Picking first matching topic.");
```
Comment (myagley) [inline edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Mqtt/MessageAddressConverter.cs]: Thanks for adding this log line.

File: edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Mqtt/MessageAddressConverter.cs [hunk 2]
```diff
@@ -86,32 +86,38 @@ public bool TryBuildProtocolAddressFromEdgeHubMessage(string endPointUri, IMessa
...
+            string address = message.Address;
+            var uri = new Uri(address, UriKind.Relative);
             IList<IList<KeyValuePair<string, string>>> matches = this.inboundTable
                 .Select(template => template.Match(uri))
                 .Where(match => match.Count > 0)
                 .ToList();
 
-            if (matches.Count > 1)
-            {
-                this.logger.LogWarning($"Message address {message.Address} matches more than one topic. Picking first matching topic.");
```
Comment (myagley) [inline edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Mqtt/MessageAddressConverter.cs]: Was this not actually being hit before then?

Comment (varunpuranik) [reply edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Mqtt/MessageAddressConverter.cs]: It was, it was just not being handled properly (was being treated as an error case)

**Comment by varunpuranik** [comment]
This PR fixes this issue - https://github.com/Azure/iotedge/issues/203

**Pull request merged**